### ISSUE TITLE
Add color output for `mage -l` targets

### ISSUE THREAD

**Issue by mirogta** [magefile/mage#285]
We now have ~30 mage build targets so the output from `mage -l` isn't very easy on eyes.

It would be nicer by coloring the build targets with some arbitrary color (cyan?) while leaving the description (comments on public mage funcs) in the default color.

Inspired by https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html

**Comment by natefinch** [comment]
Hmm.. I'm not against this, but I always worry about terminal compatibility.

**Comment by mirogta** [comment]
Fair point... Is there a way to find out all people who are currently using or looking for using Mage, and then ask them if they use a color terminal?

Would it be compatible with your terminal?

**Comment by ghostsquad** [comment]
you could look at the `TERM` env var. If it's set to something like `xterm-256color` then enable by default. You can also provide options such as

```
--color
--no-color
```

to force one behavior regardless.

https://www.gnu.org/software/gettext/manual/html_node/The-TERM-variable.html

**Comment by mirogta** [comment]
@natefinch I have a working code and a test, but not quite ready yet for a PR, because...

I have a passing test, similar to the existing `TestMageImportsList` in `import_test.go` but instead of the **expected** string with a simple string which looks like this:

```
	actual := stdout.String()
	expected := `
Targets:
  buildSubdir        Builds stuff.
  ns:deploy          deploys stuff.
  root               
  zz:buildSubdir2    Builds stuff.
  zz:ns:deploy2*     deploys stuff.

* default target
`[1:]

	if actual != expected {
		t.Logf("expected: %q", expected)
		t.Logf("  actual: %q", actual)
		t.Fatalf("expected:\n%v\n\ngot:\n%v", expected, actual)
	}
```

... I have the **expected** string with the ANSI colored text:

```
	actual := stdout.String()
	expected := `
Targets:
  \033[36mbuildSubdir%s\033[0m        Builds stuff.
  \033[36mns:deploy%s\033[0m          deploys stuff.
  \033[36mroot%s\033[0m               
  \033[36mzz:buildSubdir2%s\033[0m    Builds stuff.
  \033[36mzz:ns:deploy2*%s\033[0m     deploys stuff.

* default target
`[1:]

	if actual != expected {
		t.Logf("expected: %q", expected)
		t.Logf("  actual: %q", actual)
		t.Fatalf("expected:\n%v\n\ngot:\n%v", expected, actual)
	}

```

So that's good. The code works OK and this test is passing and I check if the terminal supports colors and don't add the colors if it doesn't.

Now the challenge comes with the existing tests, which are obviously failing, because now every test case has colorized target names.

I've thought how do I "fix" the existing tests? I wanted to ensure that the tests would run irrespective of whether the terminal running the tests supports colors or not, e.g. your own terminal may support colors, but CI with a simple output may not.

My idea is to disable the color output, i.e. override the TERM variable, for every relevant test, and enable it explicitly *only* for my test which tests the colored output. This seems the most effective and least disruptive to existing tests. So only my test would look "ugly", but that's the price for a really nice colored output. I have to say I really do like how the targets stand out when I run `mage -l`.

![image](https://user-images.githubusercontent.com/5633108/80265172-8b5fa280-868e-11ea-95d4-5a8e66fbbc8b.png)


WDYT?

**Comment by natefinch** [comment]
That looks awesome. I think overriding TERM is fine.

**Comment by mirogta** [comment]
Great - see https://github.com/magefile/mage/pull/301

**Comment by mirogta** [comment]
Closed by #301
